### ISSUE TITLE
Mounted contents files always report 0 size, maybe other missing attributes

### ISSUE THREAD

**Issue by bollwyvl** [jupyterlite/jupyterlite#671]
## Description

Some file attributes aren't getting propagated (correct or otherwise) from the contents API. This is easiest  noticeable with file sizes.

## References

- noted on #670
- follow-on to #655

## Reproduce

- upload a bunch of different kinds of files
- make a pyolite notebook 

    ```py   
    from pathlib import Path
    {p: p.stat().st_size for p in Path.cwd().glob("*")}
    ```

- all the files have a size of `0`

## Expected behavior

- files have sizes
- other attributes are correct

## Context

- JupyterLite version: #670
- Operating System and version: linux
- Browser and version: 102.0.5005.115 (Official Build) snap (64-bit)

**Comment by martinRenou** [comment]
Assigning myself on this.

**Comment by martinRenou** [comment]
So it looks like the following works-ish **if** we require the content:

https://github.com/jupyterlite/jupyterlite/blob/2974eaae5796cb7025f632b7aaa96c9799854c4f/packages/server-extension/src/index.ts#L307-L324

with:

```typescript
model = await contentManager.get(path, { content: true });
```

But the `model.size` is not set properly, it's either `undefined` or set with a weird (visibly wrong) value. I suspect this has to do with the IModel stored in the localforage.

**Comment by bollwyvl** [comment]
Yes: for example, it seems that `base64` files are always `75b`.

I guess this was less of an issue when we just had to fool the browser, but "real" code (even if running on a "fake" computer) seems to be a bit pickier... but the payoff is huge.

**Comment by martinRenou** [comment]
I wonder where this could be fixed. Should we compute the size when saving the IModel on localforage?

**Comment by jasongrout** [comment]
Possibly related to this: I'm often seeing js errors using jupyterlite 0.1.0b9 in the console of the form:

```
Uncaught (in promise) TypeError: a.path is undefined
    onmessage http://localhost:8000/lite/build/4884.7d948f6.js:1
    activate http://localhost:8000/lite/build/4884.7d948f6.js:1
    promise http://localhost:8000/lite/build/6648.62de94d.js:2
    promise callback*16731/a</e.prototype.activatePlugin http://localhost:8000/lite/build/6648.62de94d.js:2
```
Tracking it down in the debugger, it seems to come from this line:
https://github.com/jupyterlite/jupyterlite/blob/4af0da94f820730d91261db472409abb108d1068/packages/server-extension/src/index.ts#L231
In particular, the `request` variable on that line looks something like 
```
{'dev': 0,
 'ino': 0,
 'mode': 16895,
 'nlink': 0,
 'uid': 0,
 'gid': 0,
 'rdev': 0,
 'blksize': 0,
 'blocks': 0,
 'atime': '1970-01-01T00:00:00.000Z',
 'mtime': '1970-01-01T00:00:00.000Z',
 'ctime': '1970-01-01T00:00:00.000Z',
 'timestamp': 0}
```
So...are we always expecting a `.path` attribute, or should we guard against not having it?

**Comment by martinRenou** [comment]
This is odd. I guess it's a separate issue indeed. What is the Python code you are running in the Notebook?

**Comment by jasongrout** [comment]
This was in JupyterLab, and the python code was something like `1` (i.e., just entering an integer). I think it may have been the file browser's automatic request? If it sounds separate, I'll try to get more precise reproduction instructions in a new issue.

**Comment by bollwyvl** [comment]
Even the example [above](https://github.com/jupyterlite/jupyterlite/issues/671#issuecomment-1157785629) doesn't have path set.

Perhaps we route all access to the squishy `broadcast.postmessage` through something typed. Getting `services.js` to be generated from an internally-consistent `.ts` would help as well, and let us capture some more of the magic strings in a single place.

**Comment by bollwyvl** [comment]
This is probably the root cause of the new pyodide issue: https://github.com/jupyterlite/jupyterlite/pull/756#issuecomment-1210071910

**Comment by bollwyvl** [comment]
At least getting rid of `undefined` cleared up some things on #756, but non-contents IDrive stuff (a la `jupyterlab-github`) still fail pretty hard.

**Comment by martinRenou** [comment]
What do you mean by non-contents IDrive?

I might be that the JupyterLab-github extension also does not provide a size.

**Comment by bollwyvl** [comment]
Here's a concrete issue that is _probably_ caused by this: #830.

The proposed fix to `pyodide.asm.js` "fixes" it, but something we're doing is probably still not correct.

**Comment by bollwyvl** [comment]
Some progress on #839 .

**Comment by bollwyvl** [comment]
Closing this now, as the `size: 0` part is resolved, and can refine on further issues like #840
